Lock StorageObjectInUseProtection feature gate to default

ikeeip opened the pull request:
#### What type of PR is this?

/kind cleanup
/kind feature

#### What this PR does / why we need it:

Lock `StorageObjectInUseProtection` feature gate to default.

#### Which issue(s) this PR fixes:

Fixes N/A

#### Special notes for your reviewer:

Additional PR to make locking to default first before removing feature gates logic (https://github.com/kubernetes/kubernetes/pull/104903).

#### Does this PR introduce a user-facing change?

```release-note
-  Feature-gate `StorageObjectInUseProtection` has been deprecated and cannot be disabled. It will be completely removed in 1.25
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

ikeeip committed:
Lock StorageObjectInUseProtection feature gate to default

Signed-off-by: Konstantin Misyutin <konstantin.misyutin@huawei.com>

msau42 commented on pkg/features/kube_features.go:
```diff
@@ -756,7 +756,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 	CPUCFSQuotaPeriod:                              {Default: false, PreRelease: featuregate.Alpha},
 	TopologyManager:                                {Default: true, PreRelease: featuregate.Beta},
-	StorageObjectInUseProtection:                   {Default: true, PreRelease: featuregate.GA},
+	StorageObjectInUseProtection:                   {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.23
```
Following Kubernetes deprecation policy of 2 releases, this should be 1.25.

ikeeip replied:
Done

msau42 reviewed (commented):
Can you add a release note to indicate that the feature gate is deprecated?

ikeeip commented:
> Can you add a release note to indicate that the feature gate is deprecated?

But isn't this enough?

>#### Does this PR introduce a user-facing change?
>
>```release-note
>- The `StorageObjectInUseProtection` graduated to GA in v1.11 and is unconditionally enabled now.
>```

ikeeip committed:
Remove tests when StorageObjectInUseProtection feature is disabled

As well as feature gate are locked, the tests when this feature is
disabled will crash. So we should remove them together with locking
the feature.

Signed-off-by: Konstantin Misyutin <konstantin.misyutin@huawei.com>

ikeeip commented:
/retest

msau42 commented:
I didn't get any response regarding the specific deprecation wording recommendation for the release note, and it doesn't look like we've been doing this for other features, so this looks ok.

/lgtm
/approve

sftim commented:
Would it be possible to mention the words “feature gate” in the release note here?

ikeeip commented:
> Would it be possible to mention the words “feature gate” in the release note here?

Yes, fixed.

> See #105474 for an example of wording.

Thanks for the example.
